awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -94,6 +94,17 @@ def __init__(self, argument_spec, **kwargs):
 
         self.session = Request(cookies=CookieJar(), validate_certs=self.verify_ssl)
 
+    def extract_errors_from_response(self, response):
+        if 'json' in response:
+            if '__all__' in response['json']:
+                return response['json']['__all__'][0]
+            elif 'error' in response['json']:
+                return response['json']['error']
```

Comment:
what about "detail" and "msg", I think those should be inspected too because it's sometimes a thing.

Also, best to do this in a loop.

And go ahead with the duck typing, if it's a list, then make it into a string. I don't like `[0]`, I'd rather do `\n'.join()`. Your choice for the separator.

Reply:
Do you have examples of detail and msg? Are those at the root of the object returned? I'll take care of the loop and joins in the mean time.